wgpu-hal/src/auxil/mod.rs

```diff
@@ -15,6 +15,12 @@ pub mod db {
     }
 }
 
+/// Maximum binding size for the shaders that only support `i32` indexing.
+/// Interestingly, the index itself can't reach that high, because the minimum
+/// element size is 4 bytes, but the compiler toolchain still computes the
+/// offset at some intermediate point, internally, as i32.
+pub const MAX_I32_BINDING_SIZE: u32 = 1 << 31;
```

Comment:
I may miss something, but `1 << 31` does not fit in an `i32`, as the maximum value for it is `(1 << 31) - 1`.

I think that using `i32::MAX as u32` instead could be less error-prone.

Reply:
The size doesn't need to fit into i32. It's the byte offsets that are internally converted into i32, and they are strictly less than the full binding size.